### PR TITLE
potential fix for CRM-16845

### PR DESCRIPTION
**Pull request by bjendres**
proposed fix for [CRM-16845](https://issues.civicrm.org/jira/browse/CRM-16845)

### CODE REVIEW

File: CRM/Contribute/BAO/Contribution.php [hunk 1]
```diff
@@ -163,9 +163,12 @@ public static function add(&$params, $ids = array()) {
...
     $contribution->id = $contributionID;
 
-    if (!CRM_Utils_Rule::currencyCode($contribution->currency)) {
-      $config = CRM_Core_Config::singleton();
-      $contribution->currency = $config->defaultCurrency;
+    if ($contribution->id) {
+      // (only) on 'create', make sure that a valid currency is set (CRM-16845)
+      if (!CRM_Utils_Rule::currencyCode($contribution->currency)) {
+        $config = CRM_Core_Config::singleton();
+        $contribution->currency = $config->defaultCurrency;
```
Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: trivial aside you can now save a line by doing

CRM_Core_Config::singleton()->defaultCurrency;

Comment (systopia) [inline CRM/Contribute/BAO/Contribution.php]: I wanted to change as little as possible :)

Comment (eileenmcnaughton) [inline CRM/Contribute/BAO/Contribution.php]: Yeah - not a required change - just a niceity (spelling?)

**Comment by davecivicrm** [comment]
This PR did NOT fix the behavior in my test on a local 4.7 sandbox with latest pull from master branch.

**Comment by systopia** [comment]
> This PR did NOT fix the behavior in my test on a local 4.7 sandbox with latest pull from master branch.

That's weird, it did work for me... I'll double check this later today...

**Comment by systopia** [comment]
my bad, this was the wrong PR

**Pull request closed without merging**